include/drivers/gpio.h

```diff
@@ -348,15 +348,51 @@ struct gpio_dt_spec {
  *
  * @param node_id devicetree node identifier
  * @param prop lowercase-and-underscores property name
+ * @param idx logical index into "gpio_pha"
  * @return static initializer for a struct gpio_dt_spec for the property
  */
-#define GPIO_DT_SPEC_GET(node_id, prop)					\
-	{								\
-		.port = DEVICE_DT_GET(DT_GPIO_CTLR(node_id, prop)),	\
-		.pin = DT_GPIO_PIN(node_id, prop),			\
-		.dt_flags = DT_GPIO_FLAGS(node_id, prop),		\
+#define GPIO_DT_SPEC_GET_BY_IDX(node_id, prop, idx)			       \
+	{								       \
+		.port = DEVICE_DT_GET(DT_GPIO_CTLR_BY_IDX(node_id, prop, idx)),\
+		.pin = DT_GPIO_PIN_BY_IDX(node_id, prop, idx),		       \
+		.dt_flags = DT_GPIO_FLAGS_BY_IDX(node_id, prop, idx),	       \
 	}
 
+/**
+ * @brief Equivalent to GPIO_DT_SPEC_GET_BY_IDX(node_id, prop, 0).
+ *
+ * @param node_id devicetree node identifier
+ * @param prop lowercase-and-underscores property name
+ * @return static initializer for a struct gpio_dt_spec for the property
+ * @see GPIO_DT_SPEC_GET_BY_IDX()
+ */
+#define GPIO_DT_SPEC_GET(node_id, prop) \
+	GPIO_DT_SPEC_GET_BY_IDX(node_id, prop, 0)
+
+/**
+ * @brief Static initializer for a @p gpio_dt_spec from a DT_DRV_COMPAT
+ * instance's GPIO property at an index.
+ *
+ * @param inst DT_DRV_COMPAT instance number
+ * @param prop lowercase-and-underscores property name
+ * @param idx logical index into "gpio_pha"
+ * @return static initializer for a struct gpio_dt_spec for the property
+ * @see GPIO_DT_SPEC_GET_BY_IDX()
+ */
+#define GPIO_DT_SPEC_INST_GET_BY_IDX(inst, prop, idx) \
+	GPIO_DT_SPEC_GET_BY_IDX(DT_DRV_INST(inst), prop, idx)
+
+/**
+ * @brief Equivalent to GPIO_DT_SPEC_INST_GET_BY_IDX(inst, prop, 0).
+ *
+ * @param inst DT_DRV_COMPAT instance number
+ * @param prop lowercase-and-underscores property name
+ * @return static initializer for a struct gpio_dt_spec for the property
+ * @see GPIO_DT_SPEC_INST_GET_BY_IDX()
+ */
+#define GPIO_DT_SPEC_INST_GET(inst, prop) \
+	GPIO_DT_SPEC_INST_GET_BY_IDX(inst, prop, 0)
```

Comment:
Wouldn't this be more logical in this way?

```
#define GPIO_DT_SPEC_INST_GET(inst, prop) \
       GPIO_DT_SPEC_GET(DT_DRV_INST(inst), prop)
```

I know it is the same, but it seems to me more logical.

Reply:
That's a good point, I actually had the same dilemma. GPIO DT API https://github.com/zephyrproject-rtos/zephyr/blob/master/include/devicetree/gpio.h seems to follow the implemented rule. @mbolivar-nordic should be able to comment.

Reply:
I don't have strong feelings and it seems fine as is. Let's see if @mnkp as GPIO maintainer wants a change and otherwise leave as is, I think.